backend/src/ml_space_lambda/data_access_objects/group_user.py

```diff
@@ -0,0 +1,130 @@
+#
+#   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+#   Licensed under the Apache License, Version 2.0 (the "License").
+#   You may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+
+from __future__ import annotations
+
+import json
+from typing import List, Optional
+
+from dynamodb_json import json_util as dynamodb_json
+
+from ml_space_lambda.data_access_objects.dynamo_data_store import DynamoDBObjectStore
+from ml_space_lambda.enums import Permission
+from ml_space_lambda.utils.common_functions import serialize_permissions
+from ml_space_lambda.utils.mlspace_config import get_environment_variables
+
+
+class GroupUserModel:
+    def __init__(
+            self,
+            username: str,
+            group_name: str,
+            role: Optional[str] = None,
+            permissions: Optional[List[Permission]] = None,
+    ):
+        permissions = permissions if permissions else []
```

Comment:
FWIW I think in these scenarios we can just do `permissions or []` and below `role or ""` to make it slightly more readable.